Homework2

lytr777 opened the pull request:
Павленко Артём Леонидович

lytr777 committed:
ready

lytr777 committed:
Application is ready

saving state by turning
use AsyncTask

lytr777 committed:
Fix

fix double city and centering text

dtrounine commented on app/src/main/java/ru/ifmo/android_2015/citycam/CityCamActivity.java:
```diff
@@ -35,16 +71,208 @@ protected void onCreate(Bundle savedInstanceState) {
+        }
+    }
+
+    public class DownloadFileTask extends AsyncTask<Void, Integer, Long>
```
Желательно объявить этот  класс как static, чтобы он не держал неявную ссылку на внешний объект CityCamActivity

dtrounine commented on app/src/main/java/ru/ifmo/android_2015/citycam/CityCamActivity.java:
```diff
@@ -35,16 +71,208 @@ protected void onCreate(Bundle savedInstanceState) {
+                activity.cam.setText(current_cam + " of " + all_cam);
+                activity.user.setText("user : " + data[current_cam - 1].user);
+                activity.cam_id.setText("id : " + data[current_cam - 1].id);
+                Bitmap bit = decodeFile(destFile.getPath());
```
Декодировать битмап надо в фоновом потоке в методе doInBackground, потому что это долгая операция + чтение из файла.

dtrounine commented on app/src/main/java/ru/ifmo/android_2015/citycam/CityCamActivity.java:
```diff
@@ -35,16 +71,208 @@ protected void onCreate(Bundle savedInstanceState) {
+            try {
+                URL url = Webcams.createNearbyUrl(city.latitude, city.longitude,
+                        (current_cam / per_page) + 1);
+                DownloadJsonTask djt = new DownloadJsonTask(url);
```
DownloadJтsonTask никак не сохраняется при смене конфигурации, что в случае медленного интернета, когда запрос не успевает выполниться до смены конфигурации, новый таск будет запущен при повороте экрана.

Можно было бы его так же сохранять в Container

sandwwraith replied:
А в строчках
`
downloadJson();
download();
`
разве не возникнет race condition? Ведь загрузка картинки может вызваться, не дождавшись загрузки данных из Json

dtrounine commented:
ДЗ принято

lytr777 committed:
Fix

read and decode file in background process
save downloadJsonTask
fix download and downloadJson race condition

lytr777 committed:
Static

DownloadJsonTask and DownloadTask are static now
